game-core/src/main/java/games/strategy/triplea/delegate/IBattle.java

```diff
@@ -121,27 +120,29 @@ void unitsLostInPrecedingBattle(
    */
   boolean isEmpty();
 
-  /** Returns units which are dependent on the given units. */
+  /** Returns units which are dependent on the given units. Read-only. */
   Collection<Unit> getDependentUnits(Collection<Unit> units);
 
-  /** Returns units which are actually assaulting amphibiously. */
+  /** Returns units which are actually assaulting amphibiously. Read-only. */
   Collection<Unit> getAmphibiousLandAttackers();
 
-  /** Returns units which are actually bombarding. */
+  /** Returns units which are actually bombarding. Read-only. */
   Collection<Unit> getBombardingUnits();
 
-  /** Returns what round this battle is in. */
+  /** Returns what round this battle is in. Read-only. */
   int getBattleRound();
 
-  /** Returns units which are attacking. */
+  /** Returns units which are attacking. Read-only. */
   Collection<Unit> getAttackingUnits();
 
-  /** Returns units which are defending. */
+  /** Returns units which are defending. Read-only. */
```

Comment:
Nitting the 'read-only', some concerns:
- not sure it'll be clear what it means
- `Unit` is not read-only. While the returned collection is unmodifiable, the contents of the collection are not strictly read-only.

As a suggestion, a `@returns` javadoc might solve this, eg:
```
@returns Unmodifiable collection of the defending units
```

Reply:
Yeah, I was trying to use a word that wasn't immutable to try to mention that the collection is read-only. But agree that your suggested wording is even better - "Unmodifiable collection" I think is clearer, especially since that's what the API for making these uses for terminology.

Reply:
Updated comments. Thanks!